include/aie/Dialect/AIEVec/IR/AIEVecOps.td

```diff
@@ -867,4 +848,37 @@ def AIEVec_MatMulOp:
   let hasVerifier = 0;
 }
 
+def AIEVec_MulElemOp:
+  AIEVec_Op<"mul_elem", [
+    Pure, 
+    IsValidAIE2MulElemShapeAndType<"lhs", "rhs", "acc">
+  ]>,
+  Arguments<(ins AIE2MulElemLHS:$lhs,
+                 AIE2MulElemRHS:$rhs)>,
+  Results<(outs AIE2MulElemACC:$acc)> {
+  let summary = "AIE-ML vector element-wise multiply";
+  let description = [{
+    AMD-specific multiply operation that multiplies two 1-D vectors in the same channel.
+    The vector sizes are at least 512 bits.
+    `$acc = `$lhs * $rhs`.
+
+    Currently, the following are the supported type combinations:
+
+         lhs                | rhs                | Accumulator
+        :------------------:|:------------------:|:-----------------:
+         `vector<32xi8>`    | `vector<32xi8>`    | `vector<32xi8>`
+         `vector<32xi8>`    | `vector<32xi8>`    | `vector<32xi32>`
+         `vector<32xi16>`   | `vector<32xi16>`   | `vector<32xi16>`
+         `vector<32xi16>`   | `vector<32xi16>`   | `vector<32xi32>`
+         `vector<16xi32>`   | `vector<16xi32>`   | `vector<16xi32>`
+         `vector<16xbf16>`  | `vector<16xbf16>`  | `vector<16xbf16>`
+         `vector<16xbf16>`  | `vector<16xbf16>`  | `vector<16xf32>`
+         `vector<16xf32>`   | `vector<16xf32>`   | `vector<16xf32>`
```

Comment:
For `aievec.mul_elem` the accumulator is `acc32` (i32) for `i8/i16` and `acc64` (i64) for `i32`. Your Accumulator column here is more like a result type for a `arith.mulf/i` op from what you see in the e2e unit tests. To handle different result types, we introduce either `aievec.cast` or `aievec.srs` to the result of `aievec.mul_elem`. See https://github.com/Xilinx/mlir-aie/blob/aba1887b9750dea8db14fbe16417068f83a6b94c/lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp#L702-L712